app/src/main/java/com/portfolio/tasky/entry/network/EntryApiCall.kt

```diff
@@ -0,0 +1,17 @@
+package com.portfolio.tasky.entry.network
+
+import com.portfolio.tasky.entry.models.AuthenticationRequest
+import com.portfolio.tasky.entry.models.AuthenticationResponse
+import com.portfolio.tasky.entry.models.RegisterRequest
+import retrofit2.Call
+import retrofit2.http.Body
+import retrofit2.http.POST
+
+interface EntryApiCall {
+
+    @POST("/register")
+    fun registration(@Body registerModel: RegisterRequest) : Call<Void>
+
+    @POST("/login")
+    fun login(@Body authenticationModel: AuthenticationRequest) : Call<AuthenticationResponse>
```

Comment:
You don't need to use "Call"
+ if you don't have suspend keyword, then you aren't using coroutines

So:
```
@POST("/register")
suspend fun registration(@Body registerModel: RegisterRequest) : ResponseBody // This can be used if you don't want a body, but want to know if 200, 404, etc

@POST("/login")
suspend fun login(@Body authenticationModel: AuthenticationRequest): AuthenticationResponse
```

Reply:
I would love to use co-routine however my current implementation is making it tough to implement it. If you can see that when a response is received, it is passed at multiple places. Kinda like the values and value received event is spreading at various implementations. 

Ass soon as the response is received the callback sends the result to TaskyStatus and from there a loading value is passed to TaskyLoader

This helps me to receive response at multiple places and I can utilise these responses as per convenience. 

e.g. Response is sent to multiple places like TaskyStatusImpl and LoginFragment

If I have a valid response code say 200 then I won't trigger anything in TaskyStatusImpl and the received response will be processed in LoginFragment.

However if I receive code 400 or 500 then TaskyStatusImpl will show up a dialog and the result sent to LoginFragment will be null and nothing will be processed in it.

If you see the TaskyLoader is completely controlled by TaskySatus and whenever an api call begins and ends it automatically shows start loading toast and completed loading toast. Instead of this toast I will have my custom progressbar which will enable disable from the call back of TaskyStatus. This also helps me show and hide loader and also react correctly to response codes  by just adding 3 lines at required places.

Reply:
If you have all your api calls in a Repository, then the result of the api call will just be used in one place, and then the repository would be called wherever it's needed

But the important part, is that you could just have a try catch in the repository once, and it will do all the error handling there